### PR TITLE
feat: add hybridse logic op

### PR DESCRIPTION
**Pull request by aceforeverd**
close https://github.com/4paradigm/HybridSE/issues/99

+ ~~`&&`, `||` not add. Since `||` is the ANSI standard of string concat~~
+ `DIV`, add as a synonym to `/`
+ `mod` && `%`, the mod operator
+ `!` as synonym to `NOT`

`DIV` and `MOD` work as reserved keyword

create table statement
- distribution options

add `build_zetasql_parser.sh` to build parser locally

### CODE REVIEW

File: zetasql/parser/flex_tokenizer.l [hunk 1]
```diff
@@ -512,12 +515,13 @@ maxsize { return BisonParserImpl::token::KW_MAXSIZE; }
...
 message { return BisonParserImpl::token::KW_MESSAGE; }
 min { return BisonParserImpl::token::KW_MIN; }
+mod { return BisonParserImpl::token::KW_MOD; }
 model { return BisonParserImpl::token::KW_MODEL; }
 module { return BisonParserImpl::token::KW_MODULE; }
 natural { return BisonParserImpl::token::KW_NATURAL; }
 new { return BisonParserImpl::token::KW_NEW; }
 no { return BisonParserImpl::token::KW_NO; }
-not { return BisonParserImpl::token::KW_NOT; }
+{not} { return BisonParserImpl::token::KW_NOT; }
```
Comment (jingchen2222) [inline zetasql/parser/flex_tokenizer.l:524]: why `{not}`

File: zetasql/parser/flex_tokenizer.l [hunk 2]
```diff
@@ -416,6 +418,7 @@ descriptor { return BisonParserImpl::token::KW_DESCRIPTOR; }
 describe { return BisonParserImpl::token::KW_DESCRIBE; }
 deterministic { return BisonParserImpl::token::KW_DETERMINISTIC; }
 distinct { return BisonParserImpl::token::KW_DISTINCT; }
+div      { return BisonParserImpl::token::KW_DIVIDE; }
```
Comment (aceforeverd) [inline zetasql/parser/flex_tokenizer.l]: `div` 表示整除

**Comment by aceforeverd** [comment]
pls check the create table options @jingchen2222 @dl239 
https://github.com/jingchen2222/zetasql/pull/18/files?file-filters%5B%5D=.test#diff-8311ef349d01ed41bc4d46ed32d651bf1b8c3f687fab90b0945e0065cf1b25acR25

**Review by jingchen2222 — APPROVED** [review]
LGTM

File: zetasql/parser/testdata/create_table.test [hunk 3]
```diff
@@ -22,6 +22,141 @@ CREATE TABLE t1
...
+            "followers": ['fo1', 'fo2']
+        }
+    ]
+}
+''')
+==
+
+
+create table t1 (a int32, b string) options ( partitionnum = 3, replicanum = 3, distribution = [ ('le1', [ 'fo1' ]), ('le2', []), ('le3', ['fo1', 'fo2']) ] );
+--
```
Comment (aceforeverd) [inline zetasql/parser/testdata/create_table.test:114]: [ ( leader: endpoint, followers: [ endpoints]) ]

struct <string, [string]>(leader, ['follower', ''])

**Pull request merged**